scripts/cd/push-bundle-and-index-image.sh

```diff
@@ -141,11 +147,12 @@ fi
 
 echo "modifying & pushing operator index image ${INDEX_IMAGE}..."
 if [[ -n ${FROM_INDEX_IMAGE} ]] && [[ `${IMAGE_BUILDER} pull ${FROM_INDEX_IMAGE}` ]]; then
-    opm index add --bundles ${BUNDLE_IMAGE} --build-tool ${IMAGE_BUILDER} --tag ${INDEX_IMAGE} --from-index ${FROM_INDEX_IMAGE} ${PULL_TOOL_PARAM}
+    opm index add --generate --out-dockerfile index.Dockerfile --bundles ${BUNDLE_IMAGE} --build-tool ${IMAGE_BUILDER} --tag ${INDEX_IMAGE} --from-index ${FROM_INDEX_IMAGE} ${PULL_TOOL_PARAM}
 else
-    opm index add --bundles ${BUNDLE_IMAGE} --build-tool ${IMAGE_BUILDER} --tag ${INDEX_IMAGE} ${PULL_TOOL_PARAM}
+    opm index add --generate --out-dockerfile index.Dockerfile --bundles ${BUNDLE_IMAGE} --build-tool ${IMAGE_BUILDER} --tag ${INDEX_IMAGE} ${PULL_TOOL_PARAM}
 fi
 
+${IMAGE_BUILDER} build -f ./index.Dockerfile --platform ${IMAGE_PLATFORM} -t ${INDEX_IMAGE} .
```

Comment:
could you generate it either to a temp file or to some directory/file that is ignored by git?

Reply:
Creating it to a temporary file/directory it will probably mess up the build context , but I can try.
 
I can change the filename to something more custom: index.Dockerfile${TIMESTAMP} and .gitignore the filename pattern. 
I think keeping it can be useful for debugging/rebuilding just the index image.

Reply:
or just delete the file at the end of the script ;-)

Reply:
I've added temp Dockerfile creation and works.